service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/CaseProgressionReportEventHandlerTest.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.fpl.events.CaseProgressionReportEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Court;
+import uk.gov.hmcts.reform.fpl.model.email.EmailAttachment;
+import uk.gov.hmcts.reform.fpl.model.email.EmailData;
+import uk.gov.hmcts.reform.fpl.model.event.CaseProgressionReportEventData;
+import uk.gov.hmcts.reform.fpl.service.CaseProgressionReportService;
+import uk.gov.hmcts.reform.fpl.service.CourtService;
+import uk.gov.hmcts.reform.fpl.service.email.EmailService;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.enums.CaseProgressionReportType.MISSING_TIMETABLE;
+import static uk.gov.hmcts.reform.fpl.handlers.CaseProgressionReportEventHandler.FROM_EMAIL;
+import static uk.gov.hmcts.reform.fpl.utils.CsvWriter.ATTRIBUTE;
+
+@ExtendWith(MockitoExtension.class)
+class CaseProgressionReportEventHandlerTest {
+
+    @Mock
+    private EmailService emailService;
+
+    @Mock
+    private CourtService courtService;
+
+    @Mock
+    private CaseProgressionReportService caseProgressionReportService;
+
+    @InjectMocks
+    private CaseProgressionReportEventHandler caseProgressionReportEventHandler;
+
+    @Captor
+    private ArgumentCaptor<EmailData> emailDataArgumentCaptor;
+
+    @Test
+    void shouldSendNotificationWhenReportFilePresent() throws IOException {
+
+        String courtId = "344";
+        CaseProgressionReportEventData caseProgressionReportEventData = CaseProgressionReportEventData.builder()
+                .swanseaDFJCourts(courtId)
+                .reportType(MISSING_TIMETABLE)
+                .build();
+        CaseData caseDataSelected = CaseData.builder()
+                .caseProgressionReportEventData(caseProgressionReportEventData)
+                .build();
+        //todo: uncomment line below
+        //String toEmail = "test@gmail.com";
+        String toEmail = "familypubliclaw+report@gmail.com";
```

Comment:
not sure if these need to be commented out as it's just a test, I imagine it's mocked? (also in various other tests below too!)

Reply:
yeah - weirdly the test seems to pass either way!